verilog/tools/syntax/README.md

```diff
@@ -188,6 +193,45 @@ directly to a named grammar construct in the [SV-LRM]. The
 [`verilog/CST`](../../CST) library provides functions that abstract away
 internal structure.
 
+## JSON output description
+
+JSON root is an object which maps each input file name to an object containing
+parsing result for that file.
+
+### Parsing result object
+
+| Key      | Type   | Description                                              |
+|----------|--------|----------------------------------------------------------|
+| `tree`   | object | Parser tree. Present only when `--printtree` flag is specified and there were no parsing errors. |
+| `errors` | array  | List of error objects. Present only when there were any errors. |
+
+#### Parser tree object
+
+The tree consist of Node and Leaf objects.
+
+##### Node
+
+| Key        | Type   | Description                                            |
+|------------|--------|--------------------------------------------------------|
+| `tag`      | string | Node tag. See `NodeEnum` in [verilog\_nonterminals.h](../../CST/verilog_nonterminals.h) for available values. |
+| `children` | array  | List of children (Node, Leaf, or `null`).              |
+
+##### Leaf
+
+| Key            | Type   | Description                                        |
+|----------------|--------|----------------------------------------------------|
+| `start`, `end` | int    | Byte offset of symbol's first and last character in source text. |
+| `symbol`       | string | Symbol identifier.                                 |
```

Comment:
From your implementation, which calls `verilog_symbol_name`, this is more like a tag for the token type as seen by the lexer (e.g. keyword, operator, identifier), as opposed to the actual token text.  Did you mean to include both the token tag and its text?  (The leaf text would be reconstructable from the original text and this start/end byte offset range, if you care about compactness, but I imagine some consumers would rather have the token text here, even if the information is redundant.)

Reply:
It is intended to be something that identifies the token type/kind (e.g. symbol identifier, `module` keyword, decimal number). I'll update the description.
I didn't plan to include a field with token text, as it could be obtained using `start` and `end` values with one or two lines of code in most languages. Should I add this? If so, should I remove the `end` field?

I think it would be useful to include token class (e.g. keyword, operator, string). Is `GetFormatTokenType()` from `verilog/formatting/verilog_token.h` the best way to get it?

Reply:
No need to include it, as long as it was a conscious decision, but do document (this seemingly obvious statement) how to recover the original token text in the README.

Reply:
For `symbol`, maybe: "Token tag.  Enumerations come from the parser generator from %token lines in ../../parser/verilog.y"

Reply:
done